From ce5a518e9de53446f10d46fac98640f7ac026100 Mon Sep 17 00:00:00 2001
Message-Id: <ce5a518e9de53446f10d46fac98640f7ac026100.1649664169.git.geliang.tang@suse.com>
From: Ian Rogers <irogers@google.com>
Date: Wed, 13 Jan 2021 14:36:08 -0800
Subject: [PATCH] bpf, libbpf: Avoid unused function warning on
 bpf_tail_call_static
Git-commit: ce5a518e9de53446f10d46fac98640f7ac026100
Patch-mainline: v5.12-rc2
References: bsn#127
Modified-by-SEL: No


Add inline to __always_inline making it match the linux/compiler.h.
Adding this avoids an unused function warning on bpf_tail_call_static
when compining with -Wall.

Signed-off-by: Ian Rogers <irogers@google.com>
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Link: https://lore.kernel.org/bpf/20210113223609.3358812-1-irogers@google.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/lib/bpf/bpf_helpers.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/lib/bpf/bpf_helpers.h b/tools/lib/bpf/bpf_helpers.h
index 72b251110c4d..ae6c975e0b87 100644
--- a/tools/lib/bpf/bpf_helpers.h
+++ b/tools/lib/bpf/bpf_helpers.h
@@ -30,7 +30,7 @@
 #define SEC(NAME) __attribute__((section(NAME), used))
 
 #ifndef __always_inline
-#define __always_inline __attribute__((always_inline))
+#define __always_inline inline __attribute__((always_inline))
 #endif
 #ifndef __noinline
 #define __noinline __attribute__((noinline))
-- 
2.34.1

